specs/disk_details_for_hardware_inventory.adoc

```diff
@@ -0,0 +1,156 @@
+= Specification for adding disk details in node hardware inventory
+
+Add disk details also as a part of node hardware inventory
+
+== Problem description
+
+Hardware inventory details are used to show exact status of the particular
+node. Currently disk details are missing in node hardware inventory. Add disk
+details in node hardware inventory and disk details should contain device_name,
+fs_type, fs_uuid, model, mount_point, parent, size, name, disk_type, used, ssd,
+vendor, disk_id.
+
+== Use Cases
+
+Tendrl should show the current status of disk.
+
+== Proposed change
+
+* "Hdparm" is a package used to find the disk serial number and  firmware
+  version.
+* Add Hdparm as dependency.
+* From node_agent add a new function in pull_hardware_inventory to find disk
+  details and update those details into node_inventory list.
+* Modify the manager to retrieve inventory details from tendrl-node-inventory
+  file and compare it with node_inventory list, If any changes then dump the
+  disk details in etcd.
+* Store free disk details with different key and used disk details with
+  different key.
+* Create objects in tendrl definition for used disks and free disks.
+
+== Alternatives
+
+None
+
+== Data model impact
+
+* Add used disk details in etcd with key nodes/node_id/Used_disks.
+* Add free disk details in etcd with key nodes/node_id/Free_disks.
+* Add two different object in tendrl definitions for used_disk and free_disk.
+```
+     list: nodes/$Node_context.node_id/Free_disks
```

Comment:
for hwinfo command  :  hwinfo --block

22: IDE 00.0: 10600 Disk                                        
  [Created at block.245]
  Unique ID: 3OOL.qPX1W_dGFo7
  Parent ID: w7Y8.HhbUszJC6y1
  SysFS ID: /class/block/sda
  SysFS BusID: 0:0:0:0
  SysFS Device Link: /devices/pci0000:00/0000:00:1f.2/ata1/host0/target0:0:0/0:0:0:0
  Hardware Class: disk
  Model: "SAMSUNG MZ7TE512"
  Vendor: "SAMSUNG"
  Device: "MZ7TE512"
  Revision: "6L0Q"
  Driver: "ahci", "sd"
  Driver Modules: "ahci"
  Device File: /dev/sda
  Device Files: /dev/sda, /dev/disk/by-id/ata-SAMSUNG_MZ7TE512HMHP-000L1_S1GJNSAG400778, /dev/disk/by-id/wwn-0x4d30445853885002
  Device Number: block 8:0-8:15
  BIOS id: 0x80
  Drive status: no medium
  Config Status: cfg=new, avail=yes, need=no, active=unknown
  Attached to: #17 (SATA controller)

23: None 00.0: 11300 Partition
  [Created at block.434]
  Unique ID: bdUI.SE1wIdpsiiC
  Parent ID: 3OOL.qPX1W_dGFo7
  SysFS ID: /class/block/sda/sda1
  Hardware Class: partition
  Model: "Partition"
  Device File: /dev/sda1
  Device Files: /dev/sda1, /dev/disk/by-id/ata-SAMSUNG_MZ7TE512HMHP-000L1_S1GJNSAG400778-part1, /dev/disk/by-id/wwn-0x4d30445853885002-part1, /dev/disk/by-uuid/dda9f15f-c5ec-4674-894a-d9ae57b8243c
  Config Status: cfg=new, avail=yes, need=no, active=unknown
  Attached to: #22 (Disk)



how we can parse this, it not giving in any json formate